osf/migrations/0142_reinstate_path_unique_constraint.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
```

Comment:
Drive-by: This migration seems unrelated to this PR. Can you separate this into a different PR?

Reply:
The tests won't pass without the migration, so if we split them up you'll have to run the migration before this gets merged.

Reply:
Oh ok, I didn't realize this migration was related to the tests. OK, this is fine here then.

Reply:
I'm not convinced we need this migration with this PR - tests pass here https://github.com/CenterForOpenScience/osf.io/pull/8936.  We could merge that, and fix the broken data separately.